File: server/php/src/Application/Domain/Payments/PaymentDetails/RedirectPaymentMethodSpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class RedirectPaymentMethodSpecificOutput
{
    public ?string $authorisationCode = null;
    public ?CustomerBankAccount $customerBankAccount = null;
    public ?FraudResults $fraudResults = null;
    public ?string $paymentOption = null;
    public ?PaymentProduct3203SpecificOutput $paymentProduct3203SpecificOutput = null;
    public ?PaymentProduct5001SpecificOutput $paymentProduct5001SpecificOutput = null;
    public ?PaymentProduct5402SpecificOutput $paymentProduct5402SpecificOutput = null;
    public ?PaymentProduct5500SpecificOutput $paymentProduct5500SpecificOutput = null;
    public ?PaymentProduct840SpecificOutput $paymentProduct840SpecificOutput = null;
    public ?int $paymentProductId = null;
    public ?string $token = null;

    public function toArray(): array
    {
        return [
            'authorisationCode' => $this->authorisationCode,
            'customerBankAccount' => ($this->customerBankAccount ?? new CustomerBankAccount())->toArray(),
            'fraudResults' => ($this->fraudResults ?? new FraudResults())->toArray(),
            'paymentOption' => $this->paymentOption,
            'paymentProduct3203SpecificOutput' => ($this->paymentProduct3203SpecificOutput ?? new PaymentProduct3203SpecificOutput())->toArray(),
            'paymentProduct5001SpecificOutput' => ($this->paymentProduct5001SpecificOutput ?? new PaymentProduct5001SpecificOutput())->toArray(),
            'paymentProduct5402SpecificOutput' => ($this->paymentProduct5402SpecificOutput ?? new PaymentProduct5402SpecificOutput())->toArray(),
            'paymentProduct5500SpecificOutput' => ($this->paymentProduct5500SpecificOutput ?? new PaymentProduct5500SpecificOutput())->toArray(),
            'paymentProduct840SpecificOutput' => ($this->paymentProduct840SpecificOutput ?? new PaymentProduct840SpecificOutput())->toArray(),
            'paymentProductId' => $this->paymentProductId,
            'token' => $this->token,
        ];
    }

}

File: server/python/app/application/domain/payments/payment_details/redirect_payment_method_specific_output.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.customer_bank_account import CustomerBankAccount
from app.application.domain.payments.payment_details.fraud_result import FraudResults
from app.application.domain.payments.payment_details.payment_product_3203_specific_output import PaymentProduct3203SpecificOutput
from app.application.domain.payments.payment_details.payment_product_5001_specific_output import PaymentProduct5001SpecificOutput
from app.application.domain.payments.payment_details.payment_product_5402_specific_output import PaymentProduct5402SpecificOutput
from app.application.domain.payments.payment_details.payment_product_5500_specific_output import PaymentProduct5500SpecificOutput
from app.application.domain.payments.payment_details.payment_product_840_specific_output import PaymentProduct840SpecificOutput

@dataclass
class RedirectPaymentMethodSpecificOutput:
    authorisation_code: Optional[str] = None
    customer_bank_account: Optional[CustomerBankAccount] = None
    fraud_results: Optional[FraudResults] = None
    payment_option: Optional[str] = None
    payment_product_3203_specific_output: Optional[PaymentProduct3203SpecificOutput] = None
    payment_product_5001_specific_output: Optional[PaymentProduct5001SpecificOutput] = None
    payment_product_5402_specific_output: Optional[PaymentProduct5402SpecificOutput] = None
    payment_product_5500_specific_output: Optional[PaymentProduct5500SpecificOutput] = None
    payment_product_840_specific_output: Optional[PaymentProduct840SpecificOutput] = None
    payment_product_id: Optional[int] = None
    token: Optional[str] = None